synapse/cores/common.py

```diff
@@ -2384,7 +2402,63 @@ def isSetPropOk(self, prop):
         if not self.enforce:
             return True
 
-        return self.getPropDef(prop) is not None
+        pdef = self.getPropDef(prop)
+        if pdef is None:
+            return False
+
+        if not isadd and pdef[1].get('ro'):
+            return False
+
+        return True
+
+    def delTufoProp(self, tufo, name):
+        '''
+        Delete a property from a node in tufo format.
+
+        Args:
+            tufo ((str,dict)):  The node in tufo form
+            name (str): The relative property name to delete
+
+        Returns:
+            ((str,dict))    The updated node in tufo form
+
+        '''
+        form,valu = s_tufo.ndef(tufo)
+
+        prop = form + ':' + name
+
+        pdef = self.getPropDef(prop)
+
+        if pdef is not None:
+
+            # if the prop is read only, it may not be deleted
+            if pdef[1].get('ro'):
+                raise CantDelProp(name=prop, mesg='property is read only')
+
+            # if the prop has a default value, it may not be deleted
```

Comment:
10 seconds later... I think that letting the user delete the prop would lead to unintended problems later on, where other users would expect a value or defval. Also, I think that del resetting it to default probably doesn't make sense either, as the user should explicitly reset it to default.